test/PushComm/unit_tests/CreateIncentivizedChat/createIncentivizedChat.t.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import { BasePushCommTest } from "../BasePushCommTest.t.sol";
+import { Errors } from "contracts/libraries/Errors.sol";
+import { console } from "forge-std/console.sol";
+
+contract test_createIncentivizedChat is BasePushCommTest {
+    function setUp() public override {
+        BasePushCommTest.setUp();
+    }
+
+    modifier whenUserCreatesIncentivizedChat() {
+        approveTokens(actor.bob_channel_owner, address(commEthProxy), 100 ether);
+        _;
+    }
+
+    function test_WhenPassedAmountIsZero() external whenUserCreatesIncentivizedChat {
+        // it should Revert
+        changePrank(actor.bob_channel_owner);
+        vm.expectRevert(abi.encodeWithSelector(Errors.InvalidArg_LessThanExpected.selector, 1, 0));
+        commEthProxy.createIncentivizeChatRequest(actor.charlie_channel_owner, 0);
+    }
+
+    function test_WhenParametersAreCorrect() public whenUserCreatesIncentivizedChat {
+        // it should update chatInfo and call core and emit event
+        uint256 bobBalanceBefore = pushToken.balanceOf(actor.bob_channel_owner);
+        uint256 coreBalanceBefore = pushToken.balanceOf(address(coreProxy));
+
+        changePrank(actor.bob_channel_owner);
+
+        vm.expectEmit(false, false, false, true);
+        emit IncentivizeChatReqReceived(
+            actor.bob_channel_owner, actor.charlie_channel_owner, 100e18 - FEE_AMOUNT, FEE_AMOUNT, block.timestamp
+        );
+        vm.expectEmit(false, false, false, true);
+        emit IncentivizeChatReqInitiated(actor.bob_channel_owner, actor.charlie_channel_owner, 100e18, block.timestamp);
+
+        commEthProxy.createIncentivizeChatRequest(actor.charlie_channel_owner, 100e18);
+
+        assertEq(bobBalanceBefore - 100e18, pushToken.balanceOf(actor.bob_channel_owner));
+        assertEq(coreBalanceBefore + 100e18, pushToken.balanceOf(address(coreProxy)));
+
+        (address sender, uint256 time, uint256 amount) = commEthProxy.userChatData(actor.bob_channel_owner);
+        assertEq(sender, actor.bob_channel_owner);
+        assertEq(time, block.timestamp);
+        assertEq(amount, 100e18);
+    }
+
+    modifier whenCoreIsCalled() {
+        _;
+    }
+
+    function test_RevertWhen_CallerIsNotComm() external whenCoreIsCalled {
```

Comment:
Invalid Test case - Expects comm contract to call an external function which should actually be called by any user.

Refer to the review comment above on the incentivizedChatReq function for Core and fix it accordingly. cc: @0xNilesh @Zartaj0